src/main/java/frc/robot/poseestimator/OdometryData.java

```diff
@@ -0,0 +1,46 @@
+package frc.robot.poseestimator;
+
+import edu.wpi.first.math.geometry.Rotation2d;
+import edu.wpi.first.math.kinematics.SwerveModulePosition;
+
+import java.util.Optional;
+
+public class OdometryData {
+
+	private SwerveModulePosition[] wheelPositions = new SwerveModulePosition[4];
+	private Optional<Rotation2d> gyroYaw = Optional.empty();
+	private double timestamp = 0;
+
+	public OdometryData() {}
+
+	public OdometryData(SwerveModulePosition[] swerveModulePositions, Optional<Rotation2d> gyroYaw, double timestamp) {
+		this.wheelPositions = swerveModulePositions;
+		this.gyroYaw = gyroYaw;
+		this.timestamp = timestamp;
+	}
+
+	public SwerveModulePosition[] getWheelPositions() {
+		return wheelPositions;
+	}
+
+	public double getTimestamp() {
+		return timestamp;
+	}
+
+	public Optional<Rotation2d> getGyroYaw() {
+		return gyroYaw;
+	}
+
+	public void setWheelPositions(SwerveModulePosition[] wheelPositions) {
+		this.wheelPositions = wheelPositions;
+	}
+
+	public void setGyrYaw(Optional<Rotation2d> gyroYaw) {
```

Comment:
```suggestion
	public void setGyroYaw(Optional<Rotation2d> gyroYaw) {
```
broooooooooooooooooo
Also maybe call it just heading (i dont know if u care if its from the gyro or other sensor)

Reply:
what are you talking about, I clearly meant the angle is from the gyr.